OpenSim/Simulation/Model/Geometry.h

```diff
@@ -0,0 +1,659 @@
+/* -------------------------------------------------------------------------- *
+*                              OpenSim:  Geometry.h                          *
+* -------------------------------------------------------------------------- *
+* The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+* See http://opensim.stanford.edu and the NOTICE file for more information.  *
+* OpenSim is developed at Stanford University and supported by the US        *
+* National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    *
+* through the Warrior Web program.                                           *
+*                                                                            *
+* Copyright (c) 2005-2015 Stanford University and the Authors                *
+* Author(s): Ayman Habib                                                     *
+*                                                                            *
+* Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+* not use this file except in compliance with the License. You may obtain a  *
+* copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+*                                                                            *
+* Unless required by applicable law or agreed to in writing, software        *
+* distributed under the License is distributed on an "AS IS" BASIS,          *
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+* See the License for the specific language governing permissions and        *
+* limitations under the License.                                             *
+* -------------------------------------------------------------------------- */
+
+
+/*  
+ * Author:  Ayman Habib
+ */
+
+namespace OpenSim { 
+
+class PhysicalFrame;
+class Model;
+class Frame;
+class ModelComponent;
+//=============================================================================
+//=============================================================================
+/**
+ * Class Geometry is intended to be used as the base class for all
+ * geometry that needs to be represented in the system, including mesh files, 
+ * and built in analytic shapes. Any ModelComponent can specify a list of
+ * Geometry items to represent itself in graphics window. The relation between
+ * a ModelComponent and specific Geometry utilizes the Component mechanism, as
+ * the specific pieces of geometry are treated as subcomponents. The placement
+ * of the Geometry in 3D spae is computed from the Frame that the Geometry is
+ * "Connected" to.
+ *
+ * Geometry (and all its subclasses) serve as the set of higher level primitives
+ * available to OpenSim component writers to express the Geometry of interest.
+ * The Geometry class handles serialization and also the translation to a set of
+ * DecorativeGeometry objects that gets passed to the Visualization system to be
+ * rendered. 
```

Comment:
Optional style note: because of ubiquitous colorizing it doesn't really add anything now to begin every line with ` * `, and it eats up 3 characters out of our limited 80-character budget. It is also easier to reformat without dealing with the asterisks. So you can left justify block comments now after the opening `/**`.